docs/index.md

```diff
@@ -0,0 +1,1707 @@
+attestation      | No       | The attestation certificates for the corresponding pubkeys.
+
+
+Note that when provisioning a PIV token, if any of the optional fields are known,
+(e.g. `attestation` or `serial`) they should be supplied during provisioning.
+
+#### PIV token History
+
+As a failsafe measure, when a PIV token is deleted, the entry from the PIV token
+bucket is saved into a history bucket.  This bucket retains up to
+`KBMAPI_HISTORY_DURATION` days of PIV token data (see [#kbmapi-history]).
+
+The history bucket looks very similar to the PIV token bucket:
+
+```
+{
+    "desc": "token history",
+    "name": "pivtoken_history",
+    "schema": {
+        "index": {
+            "guid": { "type": "string" },
+            "cn_uuid": { "type": "uuid" },
+            "active_range": { "type": "daterange" }
+        }
+    }
+}
+```
+
+The major difference is that the index fields are not unique as well as the
+`active_range` index.  An accidentally deleted PIV token that's restored might end
+up with multiple history entries, and a CN which has had a PIV token replacement
+will also have multiple history entries.
+
+The moray entry in the history bucket also looks similar, but not quite the
+same as the PIV token bucket:
+
+```
+{
+    "active_range": "[2019-01-01 00:00:00, 2019-03-01 05:06:07]",
+    "model": "Yubico Yubikey 4",
+    "serial": 5213681,
+    "cn_uuid": "15966912-8fad-41cd-bd82-abe6468354b5",
+    "guid": "97496DD1C8F053DE7450CD854D9C95B4",
+    "pin": "123456",
+    "recovery_tokens": [{
+        "created": 123456789,
+        "token": "jmzbhT2PXczgber9jyOSApRP337gkshM7EqK5gOhAcg="
+    }, {
+        "created": 2233445566,
+        "token": "QmUgc3VyZSB0byBkcmluayB5b3VyIG92YWx0aW5l"
+    }],
+    "pubkeys": {
+       "9e": "ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYA...",
+       "9d": "ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYA...",
+       "9a": "ecdsa-sha2-nistp256 AAAAE2VjZHNhLXNoYTItbmlzdHAyNTYA..."
+    },
+    "attestation": {
+       "9e": "-----BEGIN CERTIFICATE-----....",
+       "9d": "-----BEGIN CERTIFICATE-----....",
+       "9a": "-----BEGIN CERTIFICATE-----....."
+    },
+    "comment": ""
+}
+```
+
+The major difference is the addition of the `active_range` property as well as
+the `comment` property. The `active_range` property represents the (inclusive)
+start and end dates that the provisioned PIV token was in use.
+
+It's permitted that the same provisioned PIV token might have multiple entries in
+the history table.  An example would be a PIV token accidentally deleted and
+restored would have an entry for the deletion, and then a second entry when
+the PIV token is retired (or reprovisioned).
+
+The `comment` field is an optional field that contains free form text.  It is
+intended to note the reason for the deletion.
+
+To protect the PIV token data in Moray, we will rely on the headnode disk
+encryption.
+
+**QUESTION**: Even though the HN PIV token will not use the GetTokenPin
```

Comment:
Actually, most of the Triton security relies into the admin network being unreachable. If we got a security breach that goes beyond that, Yubikey PIN retrieval wouldn't be our worse problem.

Regarding the headnode PIN, I'd say we shouldn't, since it won't be used at least until we have HN HA, which will not happen soon.